httptransport/logginghandler.go

```diff
@@ -0,0 +1,36 @@
+package httptransport
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type httpStatusWriter struct {
+	http.ResponseWriter
+	StatusCode int
+}
+
+// LoggingHandler will log HTTP requests using the pre initialized zerolog.
+func LoggingHandler(next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		log := zerolog.Ctx(r.Context())
+
+		// default HTTP StatusOK
+		lrw := &httpStatusWriter{ResponseWriter: w, StatusCode: http.StatusOK}
```

Comment:
I don't think I understand how this works, what updates lrw.StatusCode with a non http.StatusOK ?